spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/convert/TestItemWithEmbeddedEntity.java

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core.convert;
+
+import java.util.Objects;
+
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Embedded;
+
+/**
+ * @author Dmitry Solomakha
+ */
+public class TestItemWithEmbeddedEntity {
+	int intField;
+
+	@Embedded
+	EmbeddedEtity embeddedEntityField;
```

Comment:
I think we need to support collections/arrays of embedded entities because that's a feature Datastore has, but currently it looks like on both the read and write side we do the recursive call before we go into the collections.

Reply:
What if it's a tree? :-)

A collection of embedded entities makes sense, I'll create a separate issue for that.

Reply:
Good point! otherwise looks good to me